extension/json/json_functions/json_transform.cpp

```diff
@@ -58,15 +58,15 @@ static unique_ptr<FunctionData> JSONTransformBind(ClientContext &context, Scalar
 		throw InvalidInputException("JSON structure must be a constant!");
 	} else {
 		auto structure_val = ExpressionExecutor::EvaluateScalar(context, *arguments[1]);
-		if (!structure_val.DefaultTryCastAs(LogicalType::JSON)) {
+		if (!structure_val.DefaultTryCastAs(JSONCommon::JSONType())) {
```

Comment:
`DefaultTryCastAs` should never work with extension types, no? Should this not be turned into `TryCastAs`?